code/game/machinery/computer/records/medical.dm

```diff
@@ -0,0 +1,164 @@
+/obj/machinery/computer/records/medical
+	name = "medical records console"
+	desc = "This can be used to check medical records."
+	icon_screen = "medcomp"
+	icon_keyboard = "med_key"
+	req_one_access = list(ACCESS_MEDICAL, ACCESS_FORENSICS_LOCKERS)
+	circuit = /obj/item/circuitboard/computer/records/medical
+	light_color = LIGHT_COLOR_BLUE
+
+/obj/machinery/computer/records/medical/syndie
+	icon_keyboard = "syndie_key"
+
+/obj/machinery/computer/records/medical/laptop
+	name = "medical laptop"
+	desc = "A cheap Nanotrasen medical laptop, it functions as a medical records computer. It's bolted to the table."
+	icon_state = "laptop"
+	icon_screen = "medlaptop"
+	icon_keyboard = "laptop_key"
+
+	pass_flags = PASSTABLE
+	//these muthafuckas arent supposed to smooth
+	base_icon_state = null
+	smoothing_flags = NONE
+	smoothing_groups = null
+	canSmoothWith = null
+
+/obj/machinery/computer/records/medical/attacked_by(obj/item/attacking_item, mob/living/user)
+	. = ..()
+	if(!istype(attacking_item, /obj/item/photo))
+		return
+	insert_new_record(user, attacking_item)
+
+/obj/machinery/computer/records/medical/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	if(.)
+		return
+	ui = SStgui.try_update_ui(user, src, ui)
+	if (!ui)
+		character_preview_view = create_character_preview_view(user)
+		ui = new(user, src, "MedicalRecords")
+		ui.set_autoupdate(FALSE)
+		ui.open()
+
+/obj/machinery/computer/records/medical/ui_data(mob/user)
+	var/list/data = ..()
+
+	var/list/records = list()
+	for(var/datum/record/crew/target in GLOB.manifest.general)
+		var/list/notes = list()
+		for(var/datum/medical_note/note in target.medical_notes)
+			notes += list(list(
+				author = note.author,
+				content = note.content,
+				note_ref = REF(note),
+				time = note.time,
+			))
+
+		records += list(list(
+			age = target.age,
+			blood_type = target.blood_type,
+			crew_ref = REF(target),
+			dna = target.dna_string,
+			gender = target.gender,
+			major_disabilities = target.major_disabilities_desc,
+			minor_disabilities = target.minor_disabilities_desc,
+			physical_status = target.physical_status,
+			mental_status = target.mental_status,
+			name = target.name,
+			notes = notes,
+			quirk_notes = target.quirk_notes,
+			rank = target.rank,
+			species = target.species,
+		))
```

Comment:
Reminder to myself to suggest something. This feels off to me.

Reply:
Wow that is a good eye.

Reply:
notes are set in line 51, meaning line 69 is not required at all.

Reply:
I realize my conclusion that it was a recursive reference was wrong- I thought `notes` was the list being assigned on line 58 whereas it's actually `records`